pkg/queue/schedulingqueue/priority_scheduling_queue.go

```diff
@@ -79,6 +81,14 @@ func (p *PrioritySchedulingQueue) Run() {
 	go wait.Until(p.flushBackoffQCompleted, 1.0*time.Second, p.stop)
 }
 
+func (p *PrioritySchedulingQueue) GetRunStatus() bool {
+	return p.run
+}
+
+func (p *PrioritySchedulingQueue) SetRunStatus(b bool) {
+	p.run = b
```

Comment:
From the contemporary code, `b bool` is not fed as `true`. However, the generic design seems lack a lock to prevent `p.run` from being mutated concurrently.

Reply:
Solving read and write race through lock in MultiSchedulingQueue.

Reply:
Lock the outer layer of the call chain